### PR TITLE
Minor issues in the modularization

### PR DESCRIPTION
**Pull request by nikpodsh**
There is a small PRs that contains changes:
1) Creates Workflow for `modularization-main`. There were many issues with linting and testing, so it should be automated
2) The ML Studio extension shouldn't be deployed if there ML Studios are not enabled for the env
3) Fix of wrong env and dataset in share manager

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/modules/mlstudio/cdk/mlstudio_stack.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ class SageMakerDomainExtension(EnvironmentStackExtension):
     @staticmethod
     def extent(setup: EnvironmentSetup):
         _environment = setup.environment()
+        with setup.get_engine().scoped_session() as session:
```
Comment (dlpzx) [inline backend/dataall/modules/mlstudio/cdk/mlstudio_stack.py:41]: Good catch! thank you!

Comment (dlpzx) [reply backend/dataall/modules/mlstudio/cdk/mlstudio_stack.py:41]: Maybe as a future improvement idea: could we use Enums or other type for the env parameter values? Right now we need to look for the particular string in the code, both in frontend and backend

Comment (nikpodsh) [reply backend/dataall/modules/mlstudio/cdk/mlstudio_stack.py:41]: Yes, we should do something about that)

File: tests/modules/mlstudio/cdk/test_sagemaker_studio_stack.py [hunk 2]
```diff
@@ -21,6 +24,7 @@ def __init__(self, scope, id, env, **kwargs):
             **kwargs,
         )
         self._environment = env
+        self._db = db
```
Comment (dlpzx) [inline tests/modules/mlstudio/cdk/test_sagemaker_studio_stack.py:27]: This is needed for this right?
image

**Review by dlpzx — COMMENTED** [review]
Looks good, I just have one question in the tests/cdkproxy/conftest. Otherwise it looks good from my side

File: tests/cdkproxy/conftest.py [hunk 3]
```diff
@@ -43,6 +43,10 @@ def env(db, org: Organization) -> Environment:
         )
         session.add(env)
         session.commit()
+        session.add(
```
[the remark this answers is not in the document]
Comment (nikpodsh) [reply tests/cdkproxy/conftest.py:46]: No, we don't have to. We right way to do it was to define the fixture in `tests/modules/mlstudio/cdk/conftest.py` as you mentioned.
Currently, we have too many references just to take some fixtures from modules to the rest conftest files that it deserve a dedicated PR to restructure fixtures and reduce duplication.

File: tests/modules/mlstudio/cdk/test_sagemaker_studio_stack.py (hunk 2 above)
Comment (nikpodsh) [reply tests/modules/mlstudio/cdk/test_sagemaker_studio_stack.py:27]: Yep!

**Review by dlpzx — APPROVED** [review]

**Pull request merged**